File: src/main/java/org/tindalos/principle/domain/expectations/Thresholder.java
package org.tindalos.principle.domain.expectations;

public abstract class Thresholder {

	//default threshold
	protected Integer violationsThreshold = 0;
	
	public Integer getViolationsThreshold() {
		return violationsThreshold;
	}
	
	public void setViolationsThreshold(Integer violationsThreshold) {
		this.violationsThreshold = violationsThreshold;
	}
}

File: src/main/java/org/tindalos/principle/domain/expectations/Thresholder.scala
package org.tindalos.principle.domain.expectations

abstract class Thresholder(val violationsThreshold:Int = 0)

class ADP(threshold:Int = 0) extends Thresholder(threshold) {
  def this() = this(0)
}
class SDP(threshold:Int = 0) extends Thresholder(threshold){
  def this() = this(0)
}
class SAP(threshold:Int = 0, val maxDistance:Double = 0) extends Thresholder(threshold) {
  def this() = this(0)
}

